pkg/clustermesh/common/metrics.go

```diff
@@ -52,3 +54,22 @@ func MetricsProvider(subsystem string) func() Metrics {
 		}
 	}
 }
+
+// ApiserverMetrics contains metrics that should only be exported by the
+// clustermesh-apiserver or kvstoremesh.
+type ApiserverMetrics struct {
+	// InitialSyncDuration tracks the duration in seconds of the kvstore initial sync.
+	InitialSyncDuration metric.Gauge
+}
+
+func ApiserverMetricsProvider() func() ApiserverMetrics {
+	return func() ApiserverMetrics {
+		return ApiserverMetrics{
+			InitialSyncDuration: metric.NewGauge(metric.GaugeOpts{
+				Namespace: metrics.Namespace,
+				Name:      "initial_sync_time_seconds",
+				Help:      "Duration in seconds of kvstore initial sync",
```

Comment:
`kvstore` here is ambiguous; can you add a bit more context -- and reference that this is for clustermeshapi.

Reply:
The synchronization is different depending on which process it is: for clustermesh-apiserver, we're syncing resources from local cluster to the kvstore. For kvstoremesh, we're syncing the resources from remote clusters to the local kvstore. I'll try to add some additional details here without being too verbose